autobot-backend/llc/tests/test_reporting_line_routes_registered.py

```
@@ -0,0 +1,100 @@
+            continue
+        if not (m.methods & _WRITE_METHODS):
+            continue
+        checked += 1
+        dependant = getattr(m.route, "dependant", None)
+        names = {
+            getattr(d.call, "__name__", "")
+            for d in getattr(dependant, "dependencies", [])
+        }
+        if not any(
+            n.startswith("require_")
+            and "admin" in n
```

Ruff (0.16.3)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)